### PR TITLE
gh-116996: Add pystats about _Py_uop_analyse_and_optimize

### PR DESCRIPTION
**Pull request by mdboom**
* Issue: gh-116996

### CODE REVIEW

File: Python/optimizer_analysis.c [hunk 1]
```diff
@@ -412,6 +416,9 @@ optimize_uops(
 error:
     DPRINTF(3, "\n");
     DPRINTF(1, "Encountered error in abstract interpreter\n");
+    if (opcode <= MAX_UOP_ID) {
+        OPT_ERROR_IN_OPCODE(opcode);
```
Comment (Fidget-Spinner) [inline Python/optimizer_analysis.c:420]: You forgot to define this I think.

File: Python/optimizer_analysis.c [hunk 2]
```diff
@@ -412,6 +416,9 @@ optimize_uops(
 error:
     DPRINTF(3, "\n");
     DPRINTF(1, "Encountered error in abstract interpreter\n");
+    if (opcode <= MAX_UOP_ID) {
```
Comment (Fidget-Spinner) [inline Python/optimizer_analysis.c:419]: This is somewhat concerning, on one hand it's bounded by 512, then here it's bounded by MAX_UOP_ID. MAX_UOP_ID might grow to above 512, so I think bounding by either one and sticking to it is better.

**Review by Fidget-Spinner — COMMENTED** [review]
Looks good mostly, two concerns. Thanks!

File: Python/optimizer_analysis.c (hunk 1 above)
Comment (mdboom) [reply Python/optimizer_analysis.c:420]: Thanks, indeed I forgot to commit it.

File: Python/optimizer_analysis.c (hunk 2 above)
Comment (mdboom) [reply Python/optimizer_analysis.c:419]: Makes sense.

**Review by Fidget-Spinner — APPROVED** [review]
CI needs re-running due to transient failures. Let me do that.

**Pull request merged**